File: bionic-js-swift/BjsTests/Components/ToyComponent1.swift
import JavaScriptCore
import Bjs

class ToyComponent1: BjsExport {
    
    static var deallocCounter: Int = 0
    static var testId: Int?
    var testId: Int?
    
    var number1String, number2String: String?
    
    class var pi: Double? {
        get { return Double.pi }
    }
    
    class func sum(_ number1: Int?, _ number2: Int?) -> Int? {
        return number1! + number2!
    }
    
    init(_ number1: String?, _ number2: String?, _ testId: Int?) {
        self.number1String = number1
        self.number2String = number2
        self.testId = testId
    }
    
    var number1: Int? {
        get { return Int(number1String!) }
        set { number1String = String(newValue!) }
    }
    
    var number2: Int? {
        get { return Int(number2String!) }
        set { number2String = String(newValue!) }
    }
    
    func getSum(_ offset: Int?) -> Int? {
        return offset! + Int(number1String!)! + Int(number2String!)!
    }
    
    func getToySum(_ toyComponent1: ToyComponent1?) -> Int? {
        return getSum(0)! + toyComponent1!.getSum(0)!
    }
    
    deinit {
        if testId == ToyComponent1.testId {
            ToyComponent1.deallocCounter += 1
            print("**** dealloc \(ToyComponent1.deallocCounter)")
        }
    }
}

File: bionic-js-runtime-java/src/test/java/acceptance/toy1/components/ToyComponent1.java
package acceptance.toy1.components;

import acceptance.toy1.bjs.ToyComponent1BjsExport;

public class ToyComponent1 implements ToyComponent1BjsExport
{
    private String number1;
    private String number2;

    public static Double PI()
    {
        return Math.PI;
    }

    public static Integer sum(Integer number1, Integer number2) {
        return number1 + number2;
    }

    public ToyComponent1(String number1, String number2)
    {
        this.number1 = number1;
        this.number2 = number2;
    }

    @Override
    public Integer number1()
    {
        return Integer.parseInt(number1);
    }

    @Override
    public void number1(Integer value)
    {
        number1 = value.toString();
    }

    @Override
    public Integer number2()
    {
        return Integer.parseInt(number2);
    }

    @Override
    public void number2(Integer value)
    {
        number2 = value.toString();
    }

    @Override
    public Integer getSum(Integer offset)
    {
        return offset + number1() + number2();
    }

    @Override
    public Integer getToySum(ToyComponent1BjsExport toyComponent1)
    {
        return getSum(0) + toyComponent1.getSum(0);
    }
}
